### ISSUE TITLE
inspec exec on tar.gz with local library requirements doesn't work

### ISSUE THREAD

**Issue by arlimus** [inspec/inspec#779]
### Description

```
inspec exec file.tar.gz
```

with the file having `libraries/file_a.rb` containing `require 'file_b'` which is included in the archive.
### Possible Solutions

We might need to handle the internal `require` call to solve these.

**Comment by arlimus** [comment]
Conceptually we want all library files in a folder, always bundled together with the profile. Having that folder load also means that the internal require registry must (1) read from any source, not just file from disk (unlike `require`) and (2) keep track of the files it contains and the files it loads. We can take over require for all these files and defer to ruby otherwise.

**Comment by stevendanna** [comment]
I think we may have fixed this in the work for 1.0 without realizing it:

```
> cat testprofile
testprofile/        testprofile.tar.gz
cat testprofile/libraries/file_a.rb
require 'file_b'
> cat testprofile/libraries/file_b.rb
puts "Yo, I was required"
> inspec archive testprofile/
[...SNIP...]
I, [2016-09-27T16:47:50.533698 #94534]  INFO -- : Generate archive /Users/sdanna/tmp/testprofile.tar.gz.
I, [2016-09-27T16:47:50.537928 #94534]  INFO -- : Finished archive generation.
> inspec exec testprofile.tar.gz
Yo, I was required
  ✔  tmp-1.0: Create /tmp directory
     ✔  File /tmp should be directory

  File /tmp
     ✔  should be directory

Profile: InSpec Profile (testprofile)
Version: 0.1.0
Target:  local://

Profile Summary: 1 successful, 0 failures, 0 skipped
Test Summary: 2 successful, 0 failures, 0 skipped
```
